File: 프로그래머스/3/43162. 네트워크/네트워크.py
def bfs(node, computers, visited):
    visited[node] = True
    for tmp in range(len(computers)):
        if not visited[tmp] and computers[node][tmp] == 1:
            bfs(tmp, computers, visited)

def solution(n, computers):
    visited = [False] * n
    answer = 0
    
    for i in range(n):
        if not visited[i]:
            bfs(i, computers, visited)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    boolean[] visited = new boolean[201];
    
    void bfs(int i, int[][] computers){
        visited[i] = true;
        for (int j=0; j<computers[i].length; j++){
            if (visited[j] == false && computers[i][j] == 1)
                bfs(j, computers);
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        for (int i=0; i<computers.length; i++){
            if (visited[i]) continue;
            bfs(i, computers);
            answer++;
        }
        
        return answer;
        
    }
}